libreoffice-starter-extension-master/source/org/libreoffice/example/helper/FileHelper.java

```diff
@@ -0,0 +1,69 @@
+package org.libreoffice.example.helper;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import com.sun.star.deployment.PackageInformationProvider;
+import com.sun.star.deployment.XPackageInformationProvider;
+import com.sun.star.uno.Exception;
+import com.sun.star.uno.UnoRuntime;
+import com.sun.star.uno.XComponentContext;
+import com.sun.star.util.XURLTransformer;
+
+/**
+ * Helps finding file paths.
+ *
+ * @author arta.zena
+ */
+public class FileHelper {
+
+	/** Folder where XDL dialog files can be found */
+	final static String DIALOG_RESOURCES = "dialog/";
+
+	/**
+	 * Returns a path to a dialog file
+	 */
+	public static File getDialogFilePath(String xdlFile, XComponentContext xContext) {
+		return getFilePath(DIALOG_RESOURCES + xdlFile, xContext);
```

Comment:
Hm, šķiet, ka varētu arī failu ar Client-ID saglabāt pie pašiem plugina failiem. Iespējams, faila atrašanās vietu varētu noskaidrot šādi:
`FileHelper.getFilePath(".tilde-mt", xContext)`, bet tas tikai būtu iespējams, ja paplašinājums tiek ieinstalēts vietā, kur lietotājam ir rakstīšanas tiesības. Ja instalēšana notiek, piemēram "C:\Program Files", tad tā nevarēs.

Reply:
nevarēs, jo raksta program files